### PR TITLE
Update foobernetes examples to use 'policy' parameter

### PR DESCRIPTION
**Pull request by ahpook**
While debugging hiera lookups, I noticed that although
we had the load balancer policy as the single external
parameter to the workflows, it was neither defined in
the resource type nor used in the body of the workflows.

This commit adds it to the Foobernetes::Loadbalancer
struct, the test data, and the resource definitions
in the workflows themselves.

(NB foober_iter is not fixed in this commit, but it
no longer applies cleanly and I'm not sure we need to
keep it at all)

### CODE REVIEW

**Comment by thallgren** [comment]
@ahpook the reason the check fails is very likely #302 where I altered the handling of policy slightly. You'll need to rebase on master and adjust for that.

**Comment by ahpook** [comment]
Thx @thallgren - pushed a fix for the new hiera structure.

File: Makefile [hunk 1]
```diff
@@ -166,6 +166,7 @@ check-node:
 PHONY+= smoke-test
 smoke-test: lyra identity puppet-dsl yaml-dsl lyra-plugins generate
 	@echo "🔘 Running a smoke test with 'foobernetes' workflow"
+	@rm deployment.json*
```
Comment (thallgren) [inline Makefile]: This fails when the file doesn't exist (which is what's causing the travis build failure).

File: workflows/foobernetes_pp.pp [hunk 2]
```diff
@@ -17,10 +17,12 @@
     loadBalancerIP => '10.0.0.1',
     location => 'eu1',
     replica => false,
+    policy => $load_balancer_policy,
     webServerIDs => [$webserver1_id, $webserver2_id],
     tags => {
         'team' => 'lyra team',
-        'role' => 'primary'
+        'role' => 'primary',
+        'tags' => $load_balancer_policy
```
Comment (thallgren) [inline workflows/foobernetes_pp.pp]: I suggest removing the policy tag everywhere now when a proper "policy" attribute exist.

File: workflows/foobernetes.yaml [hunk 3]
```diff
@@ -98,6 +98,7 @@ steps:
       loadBalancerIP: 10.0.0.1
       location: eu1
       replica: false
+      policy: $policies.lb_policy
```
Comment (thallgren) [inline workflows/foobernetes.yaml]: I suggest removing the policy tag on lines 106 and 124 now when a proper policy attribute exists.

Comment (ahpook) [reply workflows/foobernetes.yaml]: OK, done

File: Makefile (hunk 1 above)
Comment (ahpook) [reply Makefile]: i reverted this change, it was just weirdness in my environment and not something CI will ever see.

**Comment by ahpook** [comment]
Can't believe it took me that long to track down why this failed. Removing the tags as [suggested above](https://github.com/lyraproj/lyra/pull/307#discussion_r292352626) changed the output in deployment.json and the expectation needed to be changed to match.

**Pull request merged**